monkey/monkey_island/cc/repository/i_machine_repository.py

```diff
@@ -53,6 +53,14 @@ def get_machine_by_hardware_id(self, hardware_id: HardwareID) -> Machine:
         :raises RetrievalError: If an error occurs while attempting to retrieve the `Machine`
         """
 
+    @abstractmethod
+    def get_machines(self) -> Sequence[Machine]:
+        """
+        Get all machines in the repository
+        :return: A sequence of all stored `Machine`s
+        :raises RetrievalError: If an error occurs while attempting to retrieve the `Machine`s
+        """
```

Comment:
There are small discrepancies between this code and the code on 2268-update-reporting. It's better to cherry-pick the commits you need than copy/paste. Also consider addressing the comments on that PR before cherry-picking the commits.